src/shared/dissect-image.c

```diff
@@ -2566,11 +2569,17 @@ int dissected_image_acquire_metadata(DissectedImage *m) {
         if (m->image_name) {
                 char *ext;
 
-                ext = strjoina("/usr/lib/extension-release.d/extension-release.", m->image_name, "0");
-                ext[strlen(ext) - 1] = '\0'; /* Extra \0 for NULSTR_FOREACH using placeholder from above */
+                /* Look for a named/versioned file first, then fallback to a generic extension-release
+                 * to allow images that get renamed/mangled by deployment/orchestration systems */
+                ext = strjoina("/usr/lib/extension-release.d/extension-release.",
+                               m->image_name,
+                               "0/usr/lib/extension-release.d/extension-release0");
+                /* Extra \0 for NULSTR_FOREACH using placeholders from above, as strjoin* functions would stop at \0 otherwise */
+                ext[strlen(ext) - 1] = '\0';
+                ext[STRLEN("/usr/lib/extension-release.d/extension-release.") + strlen(m->image_name)] = '\0';
                 paths[META_EXTENSION_RELEASE] = ext;
```

Comment:
What about:
```C
if (asprintf(&free_ext,
                "/usr/lib/extension-release.d/extension-release.%s%c/usr/lib/extension-release.d/extension-release%c",
                m->image_name, '\0', '\0') < 0)
           return log_oom();
paths[META_EXTENSION_RELEASE] = free_ext;
```
?

I think that's easier to read…

Reply:
sure, was trying hard to keep it in the same "style"